### PR TITLE
[Reader] Implement tags feed card actions (part 3)

### LINKED ISSUE #20670: [Reader] Implement tags feed card actions

### PR DESCRIPTION
**Pull request by RenanLukas**
Fixes #20670 

- Fix tags menu item label: "Tags" -> "Your Tags".
- Implements tags feed overflow menu UI and all actions.
- Fix "More from `<tag>`" button action: it should open the tag post list instead of filtering the tags feed.
- Implements empty state button action (no tags followed).

-----

## To Test:

- Install JP and sign in.
- Enable `reader_tags_feed` feature flag and restart the app.
- Open reader.

### Fix tags menu item label
- Open the dropdown menu.
- 🔍 Verify the tags feed item label is "Your Tags" instead of "Tags".

### Overflow menu
- Tap the tags feed item with at least one tag followed.
- 🔍 Tap the overflow menu in any post card: the items should be shown as expected. You should be able to dismiss the menu and all the actions should be working as expected (the same way they work in the other feeds).

### Fix "More from `<tag>`" button action
- 🔍 Scroll to the end of the tags horizontal list and tap the "More from `<tag>`" button: it should open the tag post list screen instead of applying the tag filter in the tags feed. Tapping the tag header should still apply the tag filter in the current feed.

image

### Empty state button action
- Unsubscribe from all your tags.
- 🔍 Verify the tags feed: you should see the empty state UI.
- 🔍 Tap the "Get Started" button: you should see the "Choose your interests" screen.
- 🔍 Select any tags and tap "Done": the tags should be seen loaded in the tags feed.
- 🔍 Verify the tags are shown in other contexts (e.g. in the "Manage Tags & Blogs" screen, in the tags feed after restarting the app).

-----

## Regression Notes

1. Potential unintended areas of impact

    - None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing and updated unit tests

3. What automated tests I added (or what prevented me from doing so)

    - Updated `ReaderTagsFeedViewModelTest.kt`.

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderTagsFeedFragment.kt [hunk 1]
```diff
@@ -263,28 +267,71 @@ class ReaderTagsFeedFragment : ViewPagerFragment(R.layout.reader_tag_feed_fragme
         }
     }
 
+    private fun observeSnackbarEvents() {
+        viewModel.snackbarEvents.observeEvent(viewLifecycleOwner) { snackbarMessageHolder ->
+            activity?.findViewById<View>(R.id.coordinator)?.let { coordinator ->
+                with(snackbarMessageHolder) {
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderTagsFeedFragment.kt]: Should we check if the fragment is attached before accessing the context here?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderTagsFeedFragment.kt [hunk 2]
```diff
@@ -263,28 +267,71 @@ class ReaderTagsFeedFragment : ViewPagerFragment(R.layout.reader_tag_feed_fragme
...
+            }
+        }
+    }
+
+    private fun observeOpenMoreMenuEvents() {
+        viewModel.openMoreMenuEvents.observe(viewLifecycleOwner) {
+            val readerCardUiState = it.readerCardUiState
+            val blogId = readerCardUiState.blogId
+            val postId = readerCardUiState.postId
+            val anchorView = binding.composeView.findViewWithTag<View>("$blogId$postId")
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderTagsFeedFragment.kt:318]: Not sure if we need to check != null here?

**Comment by daniloercoli** [comment]
> 🔍 Scroll to the end of the tags horizontal list and tap the "More from <tag>" button: it should open the tag post list screen instead of applying the tag filter in the tags feed.

This works as expected, although I found that after unsubscribing from a tag and returning to the Your Tags screen, the tag still appears in the list.

**Comment by RenanLukas** [comment]
@daniloercoli thanks for the review and for catching this bug. I've implemented a workaround to fix it. This is not the best solution since it updates the whole tags list after navigating back to Tags feed, but for a number of reasons I wasn't able to implement the ideal solution so far, which involves removing from UI the tags that were unsubscribed and adding the ones that were subscribed to instead of refreshing the whole list (thanks for the help @thomashorta ).

If the current workaround fixes the problem you've found, I propose that we implement the ideal solution in another PR so we can merge the actions ASAP allowing us to better test the project. WDYT?

**Review by daniloercoli — APPROVED** [review]
I agree with you @RenanLukas , let's proceed step by step and merge this PR for now. We should aim to implement a more refined solution before the deadline, as there are still delays in updating the UI on slow connections. 👍

**Pull request merged**